tap.d.ts

```diff
@@ -0,0 +1,39 @@
+// tap.d.ts
+
+declare module Test {
+
+  class Deferred {
+    resolve();
+    reject();
+  }
+
+  class test {
+    constructor(options?: any);
+    tearDown();
+    setTimeout(n: number);
+    endAll();
+    threw(er: Error, extra: Error, proxy: any);
+    pragma(set: any);
+    plan(n: number, comment?: string);
+    test(name: string, extra: Error, cb: () => void, deferred: Deferred);
```

Comment:
extra is a plain old bag-o-data object, not an Error.

The `deferred` is an internal-only argument, it shouldn't be exposed to the user.

The cb can return either a Promise, or nothing.